run_uh.py

```diff
@@ -277,22 +277,36 @@ def flush(self):
 
 def find_fife(paths):
 	"""Returns True if the fife module was found in one of the supplied paths."""
-	try:
-		# If FIFE can't be found then this call will throw an ImportError exception:
-		module_info = imp.find_module('fife', paths)
-		fife = imp.load_module('fife', *module_info)
+	default_sys_path = sys.path # to restore sys.path later
+	for path in paths:
+		# extract parent directory to FIFE module
+		if path.endswith("fife") and os.path.isdir(path):
+			path = os.path.dirname(path)
+		sys.path.insert(0, path)
 		try:
-			from fife import fife
-		except ImportError as e:
-			if str(e) != 'cannot import name fife':
-				log().warning('Failed to use FIFE from %s', fife)
-				log().warning(str(e))
-				if str(e) == 'DLL load failed: %1 is not a valid Win32 application.':
-					# We found FIFE but the Python and FIFE architectures don't match (Windows).
-					exit_with_error('Unsupported Python version', '32 bit FIFE requires 32 bit (x86) Python 2.')
-			return False
-	except ImportError:
-		# FIFE couldn't be found in any of the paths.
+			import fife
+			break
+		except ImportError:
+			pass
+	else:
+		log().warning("Module FIFE not found in supplied paths: {}.".format("\n".join(paths)))
+
+	# restore sys.path
+	sys.path = default_sys_path
+
+	# check if fife import was registered
+	if "fife" not in sys.modules:
+		return False
+
+	try:
```

Comment:
Not sure if I fully understand, but as far as I know, sys.modules contains the name of the module only after it has been imported. Therefore, we need to attempt to import it first :).